task_opts.go

```diff
@@ -158,7 +158,13 @@ func WithProcessKill(ctx context.Context, p Process) error {
 		return err
 	}
 	if err := p.Kill(ctx, syscall.SIGKILL, WithKillAll); err != nil {
-		if errdefs.IsFailedPrecondition(err) || errdefs.IsNotFound(err) {
+		// Kill might still return an IsNotFound error, even if it actually
+		// killed the process.
+		if errdefs.IsNotFound(err) {
+			<-s
```

Comment:
Could this also end up blocking forever? Seems like this should select on another condition, timeout or cancel

Reply:
Ya, i think just to be safe, we should change this to:

```go
select {
case <-ctx.Done():
return ctx.Err()
case <-s:
return nil
}

Reply:
Added the above suggestion.